src/vision/Offline_Filter.py

```diff
@@ -1,141 +1,154 @@
-    filtered_lines = []
-    for line in lines:
-        # Extract points from line
-        X1, Y1, X2, Y2 = line[0]
-        # Calculate slope as degrees for ease of use
-        deg_slope = degrees(np.arctan((Y2 - Y1)/(X2 - X1)))
-        if deg_slope < -40 or deg_slope > 40:
-            # Narrow slope range lines are found in
-            # Add to list of verified lined
-            filtered_lines.append(line)
-
-    # TODO: check if this and the next bunch can be combined to one step
-
-    # Check new list of lines
-    numLines = len(filtered_lines)
-    if numLines > 1:
-        # new list to eliminate duplicate lines
-        parallel = []
-        Xtotal = 0
-        Ytotal = 0
-        # Iterate and compare
-        for i in range(numLines):
-            for j in range(i + 1, numLines):
-                # Extract points from lines x2
-                iPoints = filtered_lines[i]
-                X1, Y1, X2, Y2 = iPoints[0]
-                jPoints = filtered_lines[j]
-                X3, Y3, X4, Y4 = jPoints[0]
-                # Find slopes of lines in degrees
-                iSlope = degrees(np.arctan((Y2 - Y1)/(X2 - X1)))
-                # could be moved to equals function???
-                jSlope = degrees(np.arctan((Y4 - Y3)/(X4 - X3)))
-
-                if equal(iSlope, jSlope, parallel):
-                    # Send to magic equals box
-                    # If verified, add *ONE* to final list (TODO: rename parallel, doesn't make sense)
-                    parallel.append(filtered_lines[i])
-                    # Draws lines on the empty line image, probably irrelevant, can use data points
-                    cv2.line(lineImg, (X1, Y1), (X2, Y2), (0, 255, 0), 1)
-                    # Add all X and Y values of ONE verified line
-                    Xtotal += X1
-                    Xtotal += X2
-                    # may work better alternating which set?
-                    Ytotal += Y1
-                    Ytotal += Y2
-        # Average ALL X and Y values to find midpoint of filtered lines
-        numLines = len(parallel)
-        if numLines:
-            Xcenter = Xtotal/(numLines*2)
-            Ycenter = Ytotal/(numLines*2)
-            lineImg[int(Ycenter), int(Xcenter)] = [255, 255, 255]
-
-    cv2.imshow('Lines', lineImg)
-
+MASK = cv2.inRange(HSV, LOWER_COLOR, UPPER_COLOR)
+
+# Blur image (buffer for pixel imperfections)
+MEDIAN = cv2.medianBlur(MASK, 5)
+
+# Edge detection for use in line detection
+MED_EDGES = cv2.Canny(MEDIAN, 50, 150)
+
+if debugging:
+    BLUR = cv2.GaussianBlur(MASK, (3, 3), 0)
+
+    BLUR_EDGES = cv2.Canny(BLUR, 100, 200)
+    MASK_EDGES = cv2.Canny(MASK, 100, 200)
+
+    # Empty image for drawing lines (testing)
+    FILTERED_LINE_IMG = np.zeros((HEIGHT, WIDTH, 3), np.uint8)
+    LINE_IMG = np.zeros((HEIGHT, WIDTH, 3), np.uint8)
+# Find lines in selected image
+LINES = cv2.HoughLinesP(MED_EDGES, 1, radians(0.5), 25, maxLineGap=25)
+
+# If there are lines:
+if LINES is not None:
```

Comment:
All of this line matching code should be in a helper function inside `visionfunctions.py`